modules/aws/db-agent-monitored/variables.tf

```diff
@@ -0,0 +1,74 @@
+variable "friendly_name" {
+  type        = string
+  description = "Friendly name to identify all resources"
+  validation {
+    condition     = length(var.friendly_name) >= 3
+    error_message = "Must be at least 3 characters long"
+  }
+  validation {
+    condition     = can(regex("^\\p{L}.*", var.friendly_name))
+    error_message = "Must start with a letter"
+  }
+}
+
+variable "subnet_id" {
+  type        = string
+  description = "Subnet id for the DSF MX instance"
+  validation {
+    condition     = length(var.subnet_id) >= 16 && substr(var.subnet_id, 0, 7) == "subnet-"
+    error_message = "Subnet id is invalid. Must be subnet-********"
+  }
+}
+
+variable "key_pair" {
+  type        = string
+  description = "Key pair for the ec2 instance"
+}
+
+variable "security_group_ids" {
+  type        = list(string)
+  description = "Additional Security group ids for the ec2 instance"
+  default     = []
+}
+
+variable "sg_ssh_cidr" {
+  type        = list(string)
+  description = "List of allowed ingress CIDR patterns allowing ssh protocols to the ec2 instance"
+  default     = []
+}
+
+variable "db_type" {
+  type        = string
+  default     = "PostgreSql"
+  description = "DB type provision on ec2 with an agent, available types are - ['PostgreSql', 'MySql', 'MariaDB']"
```

Comment:
Missing 'to" before provision. ec2 => EC2. Prefer : over -, for example, see error_message below.